File: src/types/file.ts
export const FILE_PARSE_STATUS_VALUES = ['pending', 'parsing', 'parsed', 'failed'] as const

export type FileParseStatus = (typeof FILE_PARSE_STATUS_VALUES)[number]

export const UPLOAD_FILE_EXTENSIONS = [
  'xlsx',
  'csv',
  'docx',
  'pdf',
  'pptx',
  'png',
  'jpg',
  'jpeg',
  'md',
  'json',
  'txt'
] as const

export const MAX_UPLOAD_FILE_SIZE_BYTES = 50 * 1024 * 1024
export const UPLOAD_FILE_ACCEPT_ATTRIBUTE = UPLOAD_FILE_EXTENSIONS.map((extension) => `.${extension}`).join(',')

export interface UploadedFile {
  created_at: string
  file_path: string
  file_size: number | null
  file_type: string
  id: string
  original_name: string
  parse_status: FileParseStatus
  parsed_content: Record<string, unknown> | null
  project_id: string
}

export interface UploadedFileListResponse {
  files: UploadedFile[]
}

export interface UploadedFileDeleteResponse {
  success: boolean
}

File: python-backend/app/schemas/file.py
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import Field

from app.schemas.base import APIModel, ORMModel
from app.schemas.enums import FileParseStatus


class UploadedFileResponse(ORMModel):
    id: str
    project_id: str
    original_name: str
    file_type: str
    file_path: str
    file_size: int | None = None
    parsed_content: dict[str, Any] | None = None
    parse_status: FileParseStatus
    created_at: datetime


class UploadedFileListResponse(APIModel):
    files: list[UploadedFileResponse] = Field(default_factory=list)


class UploadedFileDeleteResponse(APIModel):
    success: bool = True


class ParsedFileResponse(APIModel):
    file_id: str
    parsed_content: dict[str, Any] | None = None
    status: FileParseStatus
